presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchQueryBuilder.java

```diff
@@ -128,21 +138,32 @@ private static QueryBuilder buildTermQuery(BoolQueryBuilder queryBuilder, String
             }
 
             if (valuesToInclude.size() == 1) {
-                rangeQueryBuilder.must(new TermQueryBuilder(columnName, getValue(type, getOnlyElement(valuesToInclude))));
+                rangeQueryBuilder.must(new TermQueryBuilder(columnName, getValue(session, type, getOnlyElement(valuesToInclude))));
             }
             queryBuilder.should(rangeQueryBuilder);
         }
         return queryBuilder;
     }
 
-    private static Object getValue(Type type, Object value)
+    private static Object getValue(ConnectorSession session, Type type, Object value)
     {
-        if (type.equals(BIGINT) || type.equals(INTEGER) || type.equals(DOUBLE) || type.equals(BOOLEAN)) {
+        if (type.equals(BOOLEAN) ||
+                type.equals(TINYINT) ||
+                type.equals(SMALLINT) ||
+                type.equals(INTEGER) ||
+                type.equals(BIGINT) ||
+                type.equals(DOUBLE)) {
             return value;
         }
+        if (type.equals(REAL)) {
+            return Float.intBitsToFloat(((Long) value).intValue());
+        }
         if (type.equals(VARCHAR)) {
             return ((Slice) value).toStringUtf8();
         }
+        if (type.equals(TIMESTAMP)) {
+            return ISO_DATE_TIME.format(LocalDateTime.ofInstant(Instant.ofEpochMilli((Long) value), ZoneId.of(session.getTimeZoneKey().getId())));
```

Comment:
maybe format this:
```
return Instant.ofEpochMilli((Long) value)
        .atZone(ZoneId.of(session.getTimeZoneKey().getId()))
        .toLocalDateTime()
        .format(ISO_DATE_TIME);
```

Reply:
also, add `checkState(session.isLegacyTimestamp(), "new timestamp semantics not supported yet");`